securedrop/tests/pageslayout/test_source.py

```diff
@@ -131,24 +131,3 @@ def test_tor2web_warning(self):
     def test_why_journalist_key(self):
         self._source_why_journalist_key()
         self._screenshot('source-why_journalist_key.png')
-
-
-@pytest.mark.pagelayout
-class TestSourceSessionLayout(
-        functional_test.FunctionalTest,
-        source_navigation_steps.SourceNavigationStepsMixin,
-        journalist_navigation_steps.JournalistNavigationStepsMixin):
-
-    def setup(self):
-        self.session_length_minutes = 0.03
-        super(TestSourceSessionLayout, self).setup(
-            session_expiration=self.session_length_minutes)
-
-    def test_source_session_timeout(self):
-        self._source_visits_source_homepage()
-        self._source_clicks_submit_documents_on_homepage()
-        self._source_continues_to_submit_page()
-        self._source_waits_for_session_to_timeout(self.session_length_minutes)
-        self._source_enters_text_in_message_field()
-        self._source_visits_source_homepage()
-        self._screenshot('source-session_timeout.png')
```

Comment:
So while this is very similar to our functional test, it's worth keeping because the page layout test suite:

1. Is ran on translation merges in all supported locales - if we remove test coverage, if there was breakage in the exported strings/HTML placeholders from Weblate (in this case, the string indicating that the session timed out), we may not catch it for that locale.
2. Also generates screenshots for [updating the SecureDrop user guides](https://docs.securedrop.org/en/release-0.12.2/development/documentation_guidelines.html#updating-screenshots) and for context for translators in Weblate (pending task to upload those screenshots here: #3959)

(I think it is a valid point for us to combine the functional tests and the page layout tests into a single test suite with e.g. a flag for `--screenshot` instead of `--page-layout` and pass in the test locales via env var (and default to English for most developer test runs). We _can_ do that but given all the churn in the tests suites I think we should hold off on this battle for now)

Reply:
Right. Will save the pruning for a future overhaul. Restored.